src/MuiForms/BuilderImageSelector.tsx

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react'
+import { WidgetProps } from '@rjsf/core'
+import { Typography } from '@mui/material'
+import { useGetUiConfig } from '../../data/uiConfig'
+import SelectWidget from './SelectWidget'
+
+export default function BuilderImageSelector(props: WidgetProps): any {
+  const { uiConfig } = useGetUiConfig()
+  const [selectProps, setSelectProps] = React.useState(props)
+
+  React.useEffect(() => {
+    if (!uiConfig) return
+
+    const { options, value } = props
+    options.enumOptions = uiConfig.builderImages.map((version) => ({ value: version.image, label: version.name }))
+
+    setSelectProps({ ...props, value: value || options.enumOptions[0].value })
+  }, [uiConfig, props])
```

Comment:
If you destructure the props on line 7 then you won't have to depend on the whole object here and can instead depend on only `options`, `value` and `uiConfig`